RecoBTag/CTagging/interface/CharmTagger.h

```diff
@@ -0,0 +1,63 @@
+#ifndef RecoBTag_CTagging_CharmTagger_h
+#define RecoBTag_CTagging_CharmTagger_h
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "CommonTools/Utils/interface/TMVAEvaluator.h"
+#include "RecoBTau/JetTagComputer/interface/JetTagComputer.h"
+#include <mutex>
+#include "FWCore/Utilities/interface/ESInputTag.h"
+#include "RecoBTag/SecondaryVertex/interface/CombinedSVSoftLeptonComputer.h"
+#include "DataFormats/BTauReco/interface/TaggingVariable.h"
+#include "RecoBTau/JetTagComputer/interface/JetTagComputerRecord.h"
+
+//DEBUGGING
+#include "TFile.h"
+#include "TNtuple.h"
+#include <memory>
+#include <map>
+
+/** \class CharmTagger
+ *  \author M. Verzetti, U. Rochester, N.Y.
+ *  copied from ElectronTagger.h
+ */
+
+class CharmTagger : public JetTagComputer {
+public:
+  /// explicit ctor 
+	CharmTagger(const edm::ParameterSet & );
+	~CharmTagger();//{}
+  virtual float discriminator(const TagInfoHelper & tagInfo) const override;
+	virtual void initialize(const JetTagComputerRecord & record) override;
+	
+	typedef std::vector<edm::ParameterSet> vpset;
+	
+	struct MVAVar {
+		std::string name;
+		reco::btau::TaggingVariableName id;
+		size_t index;
+		bool has_index;
+		float default_value;
+	};
+
+private:
+  mutable std::mutex mutex_;
+	[[cms::thread_guard("mutex_")]] std::unique_ptr<TMVAEvaluator> mvaID_;
```

Comment:
```[[ ]]``` is C++11's new attribute syntax. An attribute is just a way to pass implementation specific information to the compiler. The static analyzer we use is based on the clang compiler and we have extended it to look for attributes which begin with ```cms::```. In this case the ```[[cms::thread_guard("mutex_")]]``` tells the static analyzer that the variable declared on that line is thread safe because it is protected from being called by multiple threads simultaneously by the variable named ```mutex_```.